src/CardDeck.php

```diff
@@ -255,11 +255,11 @@ protected function getNotifier(string $msg = null, Model\UserAction $action = nu
     /**
      * Js expression return when action afterHook executor return a Model.
      */
-    protected function jsModelReturn(Model\UserAction $action = null, string $msg = 'Done!'): array
+    protected function jsModelReturn(Model\UserAction $action = null, Model $returnEntity, string $msg = 'Done!'): array
```

Comment:
This is complex topic if UA should be immutable in sense of model of if it should be cloned when fetched on entity - related PR https://github.com/atk4/data/pull/943.

As there is no clear solution not direction now, I will accept this change now.

Reply:
actually this should use `getActionForEntity` somewhere, otherwise the UA could be for a different model

Reply:
done